src/components/shared/AutoSave/index.tsx

```diff
@@ -8,11 +8,35 @@ type AutoSaveProps = {
 };
 
 const AutoSave = ({ values, onSave, debounceDelay }: AutoSaveProps) => {
-  const debounceSave = useCallback(debounce(onSave, debounceDelay), [
-    debounceDelay,
-    onSave
-  ]);
-  useEffect(debounceSave, [debounceSave, values]);
+  // We also don't want to save when the component is unmounted, but a
+  // save had already been invoked and pending the debounce delay.
+  const isMounted = useRef(false);
+  const debouncedSave = useCallback(
+    debounce(() => {
+      // We also don't want to save when the component is unmounted, but a
+      // debounceSave had already been invoked and ispending the debounce delay.
+      if (isMounted.current) {
+        onSave();
+      }
+    }, debounceDelay),
+    []
+  );
+
+  useEffect(() => {
+    // We don't want autosave to to run on initial render because it can
+    // potentially create an empty record or update nothing.
+    if (isMounted.current) {
+      debouncedSave();
+    } else {
+      isMounted.current = true;
+    }
+  }, [values, debouncedSave]);
+
+  useEffect(() => {
```

Comment:
When the `Autosave` component unmounts (basically any page change), we need to reset the `isMounted` ref back to `false` so that the debounced save pending execution doesn't run.